addons/website_form/static/src/js/website_form_editor.js

```diff
@@ -335,8 +338,39 @@ odoo.define('website_form_editor', function (require) {
                 });
             }
 
-            return Promise.resolve(fieldRelationProm).then(function () {
-                var $content = $(qweb.render("website_form.field_" + field.type, {field: field}));
+            // Data which will be forwarded to the view
+            var data = {field: field};
+
+            // Fetch the company's currency for currency field
+            var fieldCurrencyProm;
+            var currencyId;
+            if (field.type === 'monetary') {
+                // First promise to fetch the company currency_id
+                fieldCurrencyProm = this._rpc({
+                    model: 'res.company',
+                    method: 'search_read',
+                    args: [
+                        [['id', '=', session.website_company_id]]
+                    ],
+                }).then(function (records) {
+                    currencyId = records[0].currency_id[0];
+
+                    // Second promise to fetch the currency symbol
```

Comment:
Can't you get this from the session?

https://github.com/odoo/odoo/blob/7a58931d2a6112a8455226681c453eb215c75a20/addons/web/models/ir_http.py#L40

Reply:
Seems that `request.session.uid` is falsy
[
![Screenshot from 2019-12-05 13-34-09](https://user-images.githubusercontent.com/9156538/70235789-ed614f00-1763-11ea-8160-7c58d924ea94.png)
](url)

Reply:
Maybe check why?

Reply:
The method `session_info` returns the currencies logging in but the currencies are lost after doing the flow. Anyway, this is not what I need, `session.currencies` is a dict with all active currencies. I only need the company's primary currency.